server/datastore/datastore_users.go

```diff
@@ -124,6 +124,16 @@ func testUserGlobalRole(t *testing.T, ds fleet.Datastore, users []*fleet.User) {
 		assert.Nil(t, err)
 		assert.Equal(t, user.GlobalRole, verify.GlobalRole)
 	}
+	err := ds.SaveUser(&fleet.User{
+		Name:       "some@email.asd",
+		Password:   []byte("asdasd"),
+		Email:      "some@email.asd",
+		GlobalRole: ptr.String(fleet.RoleObserver),
+		Teams:      []fleet.UserTeam{{Role: fleet.RoleMaintainer}},
+	})
+	require.IsType(t, &fleet.Error{}, err)
+	flErr := err.(*fleet.Error)
+	assert.Equal(t, flErr.Error(), "Cannot specify both Global Role and Team Roles")
```

Comment:
Ah, just noticed that you have the "expected" and "actual" values reversed for the `assert.Equal`.

The error ends up printing like:

```
            	Error:      	"{\n  \"message\": \"Cannot specify both Global Role and Team Roles\",\n  \"code\": 1\n}\n" does not contain "Error 1452: Cannot add or update a child row: a foreign key constraint fails"
```

Reply:
Ugh, this always gets me!

Reply:
me too, me too